src/sif/motorcyclecost.cc

```diff
@@ -430,6 +436,8 @@ Cost MotorcycleCost::EdgeCost(const baldr::DirectedEdge* edge,
     factor *= closure_factor_;
   }
 
+  factor *= std::pow(1.f - edge->curvature() / (kMaxCurvature + 0.1f), curvature_factor_);
```

Comment:
Yeah this is not ideal. `EdgeCost` and its reverse variant are in the routing algorithms' tightest loop and this line would get called every time since there's no if wrapping it. I think, we could pre-compute an array of factors at compile time. `curvature` on `DirectedEdge` is 4 bits wide, so there are16 unique values. If we can somehow discretize `curvature_factor` (let's say at 0.1 intervals for the sake of the argument) , we would end up with `150 possible values for factor * 16 possible values for curvature = a 2.4kB array` if I'm not mistaken. That's probably okay? We do something similar for the speed and density factors, just that the arrays are about 10x smaller.

Reply:
Can't think of a PR that introduced something like this. But basically it would look similar to this: 

https://github.com/valhalla/valhalla/blob/d7a0c877ec72602a0bd4ae3a99f03390b2c00412/valhalla/sif/dynamiccost.h#L159-L168

This is a compile time array where the discretized speed in kilometers per hour is the index into the array that holds speed factors for each speed. 

Since your implementation of curvature depends on both the edge's curvature and the user's preference (or possibly aversion) for it, this would be a two-dimensional array where, for example, the curvature is an index into the outer array and the discretized factor an index in the inner array. You could also flatten it into a single array and compute the index through simple arithmetic.

Reply:
If it helps, you can look at this PR, but it only did some cleaning. before it, those arrays were dynamically sized vectors allocated at runtime: https://github.com/valhalla/valhalla/pull/5200